modules/core/src/kmeans.cpp

```diff
@@ -458,7 +458,14 @@ double cv::kmeans( InputArray _data, int K,
         {
             best_compactness = compactness;
             if( _centers.needed() )
-                centers.copyTo(_centers);
+            {
+                Mat reshaped = centers;
+                if(_centers.rows() == 1 && _centers.channels() == dims)
```

Comment:
the fix is incorrect. The function behavior should not be dependent on the current shape of the output array. Instead, you should check whether the output array (`_centers`) has the fixed type (e.g. when `_centers` is `std::vector<someT>`) or not.

Reply:
Done.
But we need to check actual `channels()` number (which shouldn't depend on current shape of array).